lib/graphql/user.ts

```diff
@@ -0,0 +1,20 @@
+import { User as UserModel, Query } from "./schema";
+import { Model } from "./model";
+import { ResultPage } from "./resultPage";
+import { GET_USER } from "./queries";
+
+export class User extends Model<UserModel> {
+  async fetch(): Promise<ResultPage<UserModel>> {
+    throw new Error("You are allowed only to fetch your details.");
```

Comment:
Very nice, in the future this might be possible (for certain clients) so it is good to keep the same method names.